pkg/tunnelserver/peermanager.go

```diff
@@ -167,7 +168,17 @@ func (p *peerManager) addRemovePeers(endpoints *v1.Endpoints) {
 
 	toCreate, toDelete, _ := set.Diff(newSet, p.peers)
 	for _, ip := range toCreate {
-		p.server.AddPeer(fmt.Sprintf(p.urlFormat, ip), ip, p.token)
+		displayIP := ip
+		ipAddr, err := netip.ParseAddr(ip)
+		if err != nil {
+			logrus.Errorf("Unable to parse IP address %s for peer %s: %v", ip, p.server.PeerID, err)
+			continue
+		}
+		if ipAddr.Is6() {
+			logrus.Debugf("Detected ipv6 address %s for peer %s, enclosing in brackets to conform with URL formatting", ip, p.server.PeerID)
+			displayIP = fmt.Sprintf("[%s]", ip)
+		}
+		p.server.AddPeer(fmt.Sprintf(p.urlFormat, displayIP), ip, p.token)
```

Comment:
I was going to suggest using `utils.IsPlainIPV6(displayIP)` but I see that netip actually preserves whether the parsed address was an ipv4 or ipv6, good find.

Reply:
Option 2 looks good to me honestly :+1: